### PR TITLE
chore: bump to Emotion 11 in Fluent N*

### PR DESCRIPTION
**Pull request by layershifter**
#### Description of changes

This PR bumps Emotion to 11 in `@fluentui/react-northstar-emotion-renderer`, [Release notes](https://emotion.sh/docs/emotion-11).

The most important thing for us is that Emotion 11 uses Stylis v4 that has new plugin system:
- `stylis-plugin-rtl` was bumped to v2 (v1 worked for Stylis v3)
- Our custom `focusVisiblePlugin` was rewritten to support Stylis v4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -230,6 +231,15 @@
...
           "stylis"
         ]
+      },
+      {
+        "packages": [
+          "@fluentui/react-northstar-emotion-renderer"
+        ],
+        "dependencies": [
+          "@fluentui/eslint-plugin",
+          "stylis"
```
Comment (layershifter) [inline package.json:241]: As we are using Stylis v4, we can't use the same group as `@fluentui/react-northstar-fela-renderer` that uses Stylis v3...

File: packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export function createEmotionRenderer(options: CreateEmotionRendererOptions = {}
       container: target?.head,
       key: 'fui',
       nonce,
-      stylisPlugins: [focusVisiblePlugin],
+      stylisPlugins: [prefixer as StylisPlugin, focusVisiblePlugin as StylisPlugin],
```
Comment (layershifter) [inline packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx:33]: Emotion does not merge this setting, that's why we should include `prefixer` manually:

https://github.com/emotion-js/emotion/blob/3a8eaac14c1d157d0b5bb96597444e05f4c33eb1/packages/cache/src/index.js#L83

File: packages/fluentui/react-northstar-emotion-renderer/src/disableAnimations.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ const animationProps: (keyof ICSSInJSStyle)[] = [
 export function disableAnimations(styles: ICSSInJSStyle): ICSSInJSStyle {
   for (const property in styles) {
     if (animationProps.indexOf(property) !== -1) {
-      styles[property] = undefined;
+      delete styles[property];
```
Comment (layershifter) [inline packages/fluentui/react-northstar-emotion-renderer/src/disableAnimations.ts:19]: `{ animationName: undefined }` creates an empty classname 😥

File: packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx (hunk 2 above)
Comment (ling1726) [reply packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx:33]: nit: Let's store as our `defaultStylisPlugins` and reuse for the rtl case:
```suggestion
      stylisPlugins: defaultStylisPlugins,
```

File: packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx [hunk 4]
```diff
@@ -61,13 +61,14 @@ export function createEmotionRenderer(options: CreateEmotionRendererOptions = {}
     };
 
     const renderRule: RendererRenderRule = (styles, param) => {
+      const style = param.disableAnimations ? disableAnimations(styles) : styles;
```
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx:64]: Sorry I still don't understand why this needs to be hoisted

**Review by jurokapsiar — APPROVED** [review]

File: packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (layershifter) [reply packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx:64]: ### Before

```jsx
const styles = { animationName: "foo" };

if (Object.keys(styles).length === 0) {
  // non-empty, continue
  return "";
}

const style = param.disableAnimations ? disableAnimations(styles) : styles;
//                  ^ is "true"
// "style" becomes "{}" i.e. empty object
//  results in "fui-0" classname 💣
```

### After

```jsx
const styles = { animationName: "foo" };
const style = param.disableAnimations ? disableAnimations(styles) : styles;
//                  ^ is "true"
// "style" becomes "{}" i.e. empty object

if (Object.keys(style).length === 0) {
  // empty, return "" ✔
  return "";
}
```

File: packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx (hunk 2 above)
Comment (layershifter) [reply packages/fluentui/react-northstar-emotion-renderer/src/createEmotionRenderer.tsx]: Done in a66c2b5.

**Review by ling1726 — APPROVED** [review]

**Pull request merged**